starter-core/src/main/java/io/micronaut/starter/feature/dekorate/DekorateHalkyon.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.dekorate;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.LanguageSpecificFeature;
+import io.micronaut.starter.options.Language;
+
+import javax.inject.Singleton;
+
+/**
+ * Adds Dekorate Halkyon support.
+ *
+ * @author Pavol Gressa
+ * @since 2.0.1
+ */
+@Singleton
+public class DekorateHalkyon implements DekorateFeature {
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "dekorate-halkyon";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Dekorate Halkyon Support";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Generates Halkyon CRD kubernetes manifests using Dekorate Halkyon Support.";
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.CLOUD;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return applicationType == ApplicationType.DEFAULT || applicationType == ApplicationType.GRPC;
+    }
+
+    @Nullable
+    @Override
+    public String getMicronautDocumentation() {
+        return "https://micronaut-projects.github.io/micronaut-kubernetes/latest/guide/index.html";
```

Comment:
Not sure whether to link to kubernetes micronaut documentation. Strictly speaking, to generate k8s manifests, that module is not needed.

Reply:
+ this module is not now required, so either I'll add dependency on micronaut-kubernetes, or will remove this method